tests/rbd/readonly_namespace_image_migration.py

```diff
@@ -0,0 +1,162 @@
+"""Module to execute Live migration scenario on read-only namespace.
+
+Pre-requisites :
+We need cluster configured with atleast one client node with ceph-common package,
+conf and keyring files
+
+Test cases covered :
+CEPH-83573341 - Set read only on namespace and migrate the images and
+verify migration is not allowed
+
+Test Case Flow :
+1. Create pool.
+2. Create a namespace within the pool
+3. Give readonly access to the namespace
+4. Create images within the namespace
+5. Migrate the created images to other pool
+6. verify that image migration is not allowed as image is read-only.
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import Rbd, initial_rbd_config, verify_namespace_exist
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def image_migrate_and_verify(
+    rbd,
+    src_pool,
+    dest_pool,
+    namespace,
+    image,
+):
+    """Migrate the image from source to destination and verify migration
+
+    Args:
+        src_pool: source pool name
+        dest_pool: destination pool name
+        namespace: name of namespace
+    """
+    src_spec1 = src_pool + "/" + namespace + "/" + image
+    dest_spec1 = dest_pool + "/" + image
+    client_id = "client.1"
+    cmd1 = "ceph auth caps client.1"
+    cmd1 += " mon 'allow *' mds 'allow *'"
+    cmd1 += f" mgr 'profile rbd-read-only pool={src_pool}"
+    cmd1 += f" namespace={namespace}' osd 'profile rbd-read-only'"
```

Comment:
This is generic read-only client definition, can we please place it in utils ? (existing or new one)

Reply:
can you elaborate more on this.

Reply:
Can we create a method/class in rbd_utils and condense these 4 lines there ?
We need multiple client creation with some specific roles like this
we could come up with generic user management focused on rbd 

 @manasagowri Any thoughts ?

Reply:
Yes we could have a configure_client method in rbd_utils for it, take the roles to be enabled, client name, etc., as inputs and create that particular client.

Reply:
Done. 
latest log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WOUDO2